OrbitCore/LinuxPerfRingBuffer.cpp

```diff
@@ -88,19 +79,16 @@ void LinuxPerfRingBuffer::ReadHeader(perf_event_header* a_Header) {
   assert(m_Metadata->data_tail + a_Header->size <= m_Metadata->data_head);
 }
 
-//-----------------------------------------------------------------------------
 void LinuxPerfRingBuffer::SkipRecord(const perf_event_header& a_Header) {
   // write back how far we read the buffer.
   m_Metadata->data_tail += a_Header.size;
 }
 
-//-----------------------------------------------------------------------------
 void LinuxPerfRingBuffer::Read(uint8_t* a_Destination, uint64_t a_Count) {
   const uint64_t index = m_Metadata->data_tail;
   const uint32_t exponent = m_BufferLengthExponent;
 
-  // as m_BufferLength is a power of two, we can optimize index % m_BufferLength
-  // to:
+  // as m_BufferLength is a power of two, optimize index % m_BufferLength to:
```

Comment:
is there a CHECK for this?

Reply:
mmap on a perf_event_open file descriptor fails otherwise with error 22 (invalid argument), but I added an assert and changed the comment.